### ISSUE TITLE
dangerouslySetInnerHTML?

### ISSUE THREAD

**Issue by jwalton** [developit/preact#29]
In React, you can set inner HTML with something like:

    <div dangerouslySetInnerHTML={{__html: 'my markup'}} />

but preact doesn't seem to support this?  Any tips on how to achieve something similar?

**Comment by developit** [comment]
This is intentional ly not in preact, though it could be added to preact-compat. You can set innerHTML in a life cycle method like `componentDidUpdate` or `componentDidMount`.

Example:

```js
class Foo extends Component {
  componentDidMount() {
    this.base.innerHTML = 'hello';
  }
}
```

**Comment by jwalton** [comment]
That works client side, but obviously would not work for server side rendering (since componentDidMount() will never run on the server.)

(Apologies if this comment shows up twice - I replied by email but github seems to have lost the email version. :P)

**Comment by developit** [comment]
True, very good point. Perhaps that's the selling point of `__dangerouslySetInnerHTML` then!

**Comment by developit** [comment]
@jwalton - just published a prerelease of 3.0 which supports `dangerouslySetInnerHTML`. On npm as  `preact-3.0.0-beta1`

**Comment by jwalton** [comment]
You, sir, are a scholar and a gentleman.  :)

**Comment by developit** [comment]
Now generally available as of `3.2`.
